src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java

```diff
@@ -0,0 +1,255 @@
+    assertThat(actualJsonArray.size()).isEqualTo(expectedArrayList.size());
+  }
+
+  @Test
+  public void doPost_persistDataInDatastoreForLongVideo() throws ServletException, IOException {
+    when(request.getParameter(TranscriptServlet.PARAM_VIDEO_ID)).thenReturn(LONG_VIDEO_ID);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_C);
+
+    servlet.doPost(request, response);
+    int actualQueryCount = countEntitiesInDatastore(LECTURE_ID_C);
+    int expectedQueryCount = (extractJsonAsArrayList(LONG_VIDEO_JSON)).size();
+
+    assertThat(actualQueryCount).isEqualTo(expectedQueryCount);
+  }
+
+  @Test
+  public void doGet_OnlyOtherLecturesInDatastore_GetNoLectures()
+      throws ServletException, IOException {
+    putJsonInDatastore(SHORT_VIDEO_JSON, LECTURE_ID_B);
+    putJsonInDatastore(LONG_VIDEO_JSON, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_C);
+
+    servlet.doGet(request, response);
+    String actualJson = content.toString();
+    List<TranscriptLine> expectedArrayList = new ArrayList<>();
+    List<TranscriptLine> actualJsonArray = extractJsonAsArrayList(actualJson);
+
+    assertThat(actualJsonArray).isEqualTo(expectedArrayList);
+  }
+
+  @Test
+  public void doGet_TwoLecturesInDatastore_GetOneLecture() throws ServletException, IOException {
+    putJsonInDatastore(SHORT_VIDEO_JSON, LECTURE_ID_B);
+    putJsonInDatastore(LONG_VIDEO_JSON, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doGet(request, response);
+    String actualJson = content.toString();
+    List<TranscriptLine> expectedArrayList = extractJsonAsArrayList(LONG_VIDEO_JSON);
+    List<TranscriptLine> actualJsonArray = extractJsonAsArrayList(actualJson);
+
+    assertThat(actualJsonArray.size()).isEqualTo(expectedArrayList.size());
+  }
+
+  private List<TranscriptLine> extractJsonAsArrayList(String json) {
+    return (ArrayList<TranscriptLine>) gson.fromJson(
+        json, (new ArrayList<List<TranscriptLine>>().getClass()));
+  }
+
+  private void putJsonInDatastore(String json, String lectureKeyId) {
+    List<TranscriptLine> transcriptLineArray = extractJsonAsArrayList(json);
+    Key lectureKeyKey =
+        KeyFactory.createKey(TranscriptServlet.PARAM_LECTURE, Long.parseLong(lectureKeyId));
+    for (int i = 0; i < transcriptLineArray.size(); i++) {
+      Entity lineEntity = new Entity(TranscriptLine.ENTITY_KIND);
+      lineEntity.setProperty(TranscriptLine.PROP_LECTURE, lectureKeyKey);
+      // Set dummy values because AutoValue needs all the values to create a TranscriptLine object.
+      lineEntity.setProperty(TranscriptLine.PROP_START, "");
+      lineEntity.setProperty(TranscriptLine.PROP_DURATION, "");
+      lineEntity.setProperty(TranscriptLine.PROP_CONTENT, "");
+      datastore.put(lineEntity);
+    }
+  }
+
+  private int countEntitiesInDatastore(String lectureKeyId) {
```

Comment:
nit: I would rename this "entitiesInDatastoreCount" the ordering of the word count implies you will get back the count.

Reply:
got it! done!